files/build_deploy.py

```diff
@@ -206,9 +213,9 @@ def notify_error(stage, error):
         notify_error('install', C.output)
 
     try:
-        syslog.syslog("Build of {}: bundle exec middleman build".format(name))
-        result = subprocess.check_output(['bundle', 'exec', 'middleman',
-                                          'build', '--verbose'])
+        command = builder_commands[config['builder']]
```

Comment:
I think some well-documented magic for simple things not requiring customization if nice. Gemfile->bundler is a simple one, no parameters needed, always the same command to run.

nevertheless there are tasks which are custom, like rh-events runs a dedicated ruby script. We may have ordering problems too (like do something before the build but after bundler).

So I guess we can defined these phases which would be enough for all cases I can think of in our current work:

- deps installation: currently bundler, maybe other things later
- pre-build hook, if exists
- build: run middleman or any method we know
- post-build hook, if exists

for maximum flexibility we could allow to not run the magic build.

maybe we could even have a single YAML file with just a list of commands to run, and have special magic command names for the smart deps install and build phase, so you could reorder or skip very easily. An missing recipe file would result into chaining these two magic actions.

```
---
recipe:
  - ./generate_ruby_deps.pl
  - AUTO_DEPS_INSTALL
  - ./patch_broken_ruby_gem.rb
  - AUTO_BUILD
  - ./cleanup_nonfree.py
  - find . -name \*.py -exec pep8 --ignore=E501,E402 {} +
```